mso/provider_test.go

```diff
@@ -84,3 +86,32 @@ func testCheckResourceDestroyPolicy(s *terraform.State, resource, policyType str
 	}
 	return nil
 }
+
+func testAccVerifyKeyValue(resourceAttrsMap *map[string]string, resourceAttrRootkey, stateKey, stateValue string) {
+	stateKeySplit := strings.Split(stateKey, ".")
+	for inputKey, inputValue := range *resourceAttrsMap {
+		if strings.Contains(stateKey, resourceAttrRootkey) && stateKeySplit[len(stateKeySplit)-1] == inputKey && (stateValue == inputValue || (inputValue == "reference" && stateValue != "")) {
```

Comment:
Shall we change this:
(isReference(inputValue) && stateValue != "")
where IsReference () checks if the value has .?